dldashboard/web/design_doc.md

```diff
@@ -0,0 +1,63 @@
+# Design document for dashboard
+
+## Architecture 
+A high level design diagram of the current architecture of the dashboard is :
+
+![dashboard architecture diagram](https://craftassist.s3-us-west-2.amazonaws.com/pubr/dashboard_architecture.png)
+
+In the diagram above:
+- `**frontend**`: The React based frontend that hosts other React subcomponents in a given layout. All the components are expected to reflect the most up-to-date state whenever there is a state change communicated from backend. 
+- `**StateManager**`: The module that manages messaging between `frontend` and `backend`. Through the StateManager : we set the state of the React components with the data communicated from backend, and the components can explicitly request information / set the state of backend. The `StateManager` and the `backend` communicate wiht each other using `socket.io` connections. The goal is to have `StateManager` be completely in sync with the backend but we aren't there yet.
+- `**backend**` : This is the agent process that sends and receives data from the `frontend` through the `StateManager` component. The agent process and its modules (perception, memory, mover, controller etc) can communicate with `StateManager` using `socket.io` connections. 
+
+
+## Snapshot 
+Here's a snapshot of how the current dashboard looks like :
+![dashboard screenshot](https://craftassist.s3-us-west-2.amazonaws.com/pubr/dashboard_screenshot.png)
+
+On a high level, following is the layout in the source code :
+- `Index.js` is the main React component that hosts other React Components in the following layout :
+  - Left:
+    - `MainPane.js`:
+      - Top:
+        - `InteractApp`: Lets you send a command to the assistant using voice or text. The commands once sent, are clickable to annotate if there was a language understanding error. This component also shows the status of the command execution by displaying agent reply or agent error messages.
+      - Bottom:
+        - `LiveImage` - Shows the rgb image stream
+        - `LiveImage` - Shows the depth map
+        - `LiveObjects` - Shows detector output
+        - `LiveHumans` - Shows human pose estimation
+  - Right:
+    - Top right:
+      - `Memory2D`: Visualizes the memory map on a 2D grid.
+      - `History`: Shows the past 5 chat interactions through the dashboard.
+      - Query the Semantic Parser (`QuerySemanticParser`): Lets you query the semantic parser with some text.
+      - Program The Assistant (`TeachApp`): Lets you do construct new programs for the assistant by assembling together blockly like blocks.
+    - Bottom right:
+      - `Settings`: Shows current settings, connection status, image quality, resolution etc
+      - `Navigator`: Lets you navigate the assistant through keypress (up, down, left, right)
+      - `ObjectFixUp`: Lets you annotate objects in dashboard.
```

Comment:
should have explained my initial comment on this better - my thought was that all of this text (shortened) could go in the image annotations since the image has so much real estate, and we could get rid of this text section entirely.